### PR TITLE
Master to trunk: secondary renames (docs and comments)

### PR DESCRIPTION
**Pull request by coder-karen**
#### Changes proposed in this Pull Request:

* Changes in this Pull Request are either in comments or docs. Where this includes links, there have been some links updated that either weren't working or references the wrong default branch in third party repositories.

* The changes here should complement (and be merged after) the changes in this PR: https://github.com/Automattic/jetpack/pull/24661

#### Other notes

*  In [monorepo.md](https://github.com/Automattic/jetpack/blob/master/docs/monorepo.md#php-tests)  under 'PHP tests for plugins' when talking about WP_BRANCH, it appears that should already be referencing trunk based on [setup-wordpress-env.sh](https://github.com/Automattic/jetpack/blob/master/.github/files/setup-wordpress-env.sh) in `.github/files`. That file give three cases - 'trunk', 'latest' and 'previous'. I've added those changes to this PR anyway.
* In `docs/release-management.md` I changed 'master issue' to 'primary issue' purely because it was already referenced as 'primary issue' in one place in the doc as well - plus this: p1HpG7-9xK-p2

#### Jetpack product discussion

p9dueE-59T-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Proof-read - all you should see is changes from master to trunk or corrected links, nothing different added.
* This PR is not for merging. It should be the second PR to be merged after both the mirror repos and the main Jetpack monorepo have had their default branches renamed.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/README.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ If your block depends on another block, place them all in extensions folder:
 
 ### High-level overview of the development flow
 
-1. Use the [Jetpack Docker environment](https://github.com/Automattic/jetpack/tree/master/docker#readme).
+1. Use the [Jetpack Docker environment](https://github.com/Automattic/jetpack/tree/trunk/docker#readme).
```
Comment (anomiex) [inline projects/plugins/jetpack/extensions/README.md:64]:
```suggestion
1. Use the [Jetpack Docker environment](https://github.com/Automattic/jetpack/tree/trunk/tools/docker#readme).
```

File: projects/plugins/jetpack/extensions/README.md [hunk 2]
```diff
@@ -142,7 +142,7 @@ Note that adding [Jest snapshot tests](https://jestjs.io/docs/en/snapshot-testin
 
 Yes! Just like any other changes in Jetpack, also blocks work in Jurassic Ninja.
 
-Simply add branch name to the URL: jurassic.ninja/create/?jetpack-beta&branch=master or use other ninjastic features.
+Simply add branch name to the URL: jurassic.ninja/create/?jetpack-beta&branch=trunk or use other ninjastic features.
```
Comment (anomiex) [inline projects/plugins/jetpack/extensions/README.md:145]:
```suggestion
Simply add branch name to the URL: jurassic.ninja/create/?jetpack-beta&branches.jetpack=trunk or use other ninjastic features.
```

File: tools/docker/README.md [hunk 3]
```diff
@@ -73,7 +73,7 @@ You can set the following variables on a per-command basis (`PORT_WORDPRESS=8000
 
 ### Container Environments
 
-Configurable settings are documented in the [`./tools/docker/default.env` file](https://github.com/Automattic/jetpack/blob/master/docker/default.env).
+Configurable settings are documented in the [`./tools/docker/default.env` file](https://github.com/Automattic/jetpack/blob/trunk/docker/default.env).
```
Comment (anomiex) [inline tools/docker/README.md:76]:
```suggestion
Configurable settings are documented in the [`./tools/docker/default.env` file](https://github.com/Automattic/jetpack/blob/trunk/tools/docker/default.env).
```

**Review by anomiex — COMMENTED** [review]
I see two URLs were missed:

* https://github.com/Automattic/jetpack/blob/b9eb81732939dcf184fbb8053366f255115b2660/projects/plugins/jetpack/_inc/lib/debugger/class-jetpack-cxn-tests.php#L920 → https://github.com/Automattic/jetpack/blob/trunk/projects/packages/sync/src/class-settings.php
* https://github.com/Automattic/jetpack/blob/b9eb81732939dcf184fbb8053366f255115b2660/readme.md?plain=1#L4 → https://codecov.io/gh/Automattic/jetpack/branch/trunk/graph/badge.svg?token=GmiI6Xl10U

File: projects/github-actions/push-to-mirrors/README.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ directory to multiple GitHub mirror repositories.
 name: Build
 on:
   push:
-    branches: [ master ]
+    branches: [ trunk ]
```
[the remark this answers is not in the document]
Comment (coder-karen) [reply projects/github-actions/push-to-mirrors/README.md:12]: Oh no, I see now that of all files it was the READMEs you mentioned in particular. Thanks for catching and correcting them.

File: projects/plugins/boost/docs/DEVELOPEMENT_GUIDE.md [hunk 5]
```diff
@@ -146,9 +146,9 @@ class Foo extends State {
 
 3.  Add `foo` to the array in `ENABLED_MODULES_DEFAULT` method in `class-jetpack-boost.php` to make it enabled by default if wanted.
 
-Every available modules are getting [instanciated](https://github.com/Automattic/jetpack/blob/master/projects/plugins/boost/app/class-jetpack-boost.php#L234) in the [prepare_modules](https://github.com/Automattic/jetpack/blob/master/projects/plugins/boost/app/class-jetpack-boost.php#L234) method of the [Jetpack Boost class](https://github.com/Automattic/jetpack/blob/master/projects/plugins/boost/app/class-jetpack-boost.php).
+Every available modules are getting [instanciated](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/boost/app/class-jetpack-boost.php#L234) in the [prepare_modules](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/boost/app/class-jetpack-boost.php#L234) method of the [Jetpack Boost class](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/boost/app/class-jetpack-boost.php).
```
[the remark this answers is not in the document]
Comment (coder-karen) [reply projects/plugins/boost/docs/DEVELOPEMENT_GUIDE.md:149]: Good idea, I'll make a note to follow up (also 'DEVELOPEMENT_GUIDE.md' itself needs it's spelling fixed).

File: projects/plugins/jetpack/_inc/client/README.md [hunk 6]
```diff
@@ -11,7 +11,7 @@ It is preferred to implement all new components as [function components](https:/
 
 ### Data approach on the Admin Page
 
-The **Admin Page** uses **redux**, [redux-thunk](https://github.com/gaearon/redux-thunk), and [react-redux](https://github.com/reactjs/react-redux)  for state handling trying to ressemble [Calypso's data approach, third Era](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#third-era-redux-global-state-tree-december-2015---present).
+The **Admin Page** uses **redux**, [redux-thunk](https://github.com/gaearon/redux-thunk), and [react-redux](https://github.com/reactjs/react-redux)  for state handling trying to ressemble [Calypso's data approach, third Era](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md#third-era-redux-global-state-tree-december-2015---present).
```
[the remark this answers is not in the document]
Comment (coder-karen) [reply projects/plugins/jetpack/_inc/client/README.md:14]: Good catch. I'll commit separately so I can also fix 'ressemble' in that sentence.

**Comment by coder-karen** [comment]
> `https://codecov.io/gh/Automattic/jetpack/branch/master/graph/badge.svg?token=GmiI6Xl10U`

I had looked at this one but couldn't find the equivalent in the monorepo so assumed the setup was different there. It sounds like the URL also changes with our default branch rename so it makes sense to change it.

> `https://github.com/Automattic/jetpack/blob/master/packages/sync/src/class-settings.php`

Oh good catch! I see how I missed that one with all the master user references in the file. Got that one too now, and also corrected the path.

Speaking of links, thank you for all the other catches you made in this PR with links. I've gone through another round as well and found even more.
Also for your catches with regard to the accuracy of some of the existing text and for the suggestions there. It's definitely a good idea to be checking for general accuracy of the text, I appreciate that.

File: projects/plugins/jetpack/_inc/client/components/jetpack-notices/README.md [hunk 7]
```diff
@@ -24,16 +24,16 @@ Remove Notice
 
 [Read the full readme](../../notices/README.md)
 
-## State Notices [#](https://github.com/Automattic/jetpack/blob/master/_inc/client/components/jetpack-notices/state-notices.jsx)
+## State Notices [#](https://github.com/Automattic/jetpack/blob/trunk/_inc/client/components/jetpack-notices/state-notices.jsx)
 
 These are triggered by PHP, most commonly to display an error during connection.
 
-- These notices are triggered by a cookie set in the browser in PHP, by the `Jetpack::state()` [#](https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L4661) method.
+- These notices are triggered by a cookie set in the browser in PHP, by the `Jetpack::state()` [#](https://github.com/Automattic/jetpack/blob/trunk/class.jetpack.php#L4661) method.
```
[the remark this answers is not in the document]
Comment (anomiex) [reply projects/plugins/jetpack/_inc/client/components/jetpack-notices/README.md:31]: Works for me

File: projects/plugins/beta/docs/testing/testing-tips.md [hunk 8]
```diff
@@ -8,7 +8,7 @@ To test out Jetpack, we recommend setting up a publicly accessible test site. Te
 
 **Jetpack Live Branches** is a [Tampermonkey](https://tampermonkey.net/) script that allows us to launch WordPress Sites with a Jetpack version coming from a changeset present on a Pull Request. This is achieved by installing the [Jetpack Beta Tester plugin](https://github.com/Automattic/jetpack-beta) on a fresh new WordPress site and switching to the branch matching the Pull Request.
 
-More details on how to use in the [Jetpack Live Branches README](https://github.com/Automattic/jetpack/blob/trunk/tools/jetpack-live-branches/README.md#readme).
+More details on how to use in the [Jetpack Live Branches README](https://github.com/Automattic/jetpack/blob/trunk/tools/jetpack-live-branches/README.md).
```
Comment (anomiex) [inline projects/plugins/beta/docs/testing/testing-tips.md:11]: Might want to keep the anchor here. It tries to skip past all GitHub's headers to get to the actual document.
```suggestion
More details on how to use in the [Jetpack Live Branches README](https://github.com/Automattic/jetpack/blob/trunk/tools/jetpack-live-branches/README.md#readme).
```

**Review by anomiex — COMMENTED** [review]
> I had looked at this one but couldn't find the equivalent in the monorepo so assumed the setup was different there. It sounds like the URL also changes with our default branch rename so it makes sense to change it.

It's one of those cases where you could put any branch name into the URL (or at least any that codecov knows about). For example, https://codecov.io/gh/Automattic/jetpack/branch/jetpack/branch-11.0/graph/badge.svg?token=GmiI6Xl10U shows the coverage for the jetpack/branch-11.0 release branch.

>  Also for your catches with regard to the accuracy of some of the existing text and for the suggestions there. It's definitely a good idea to be checking for general accuracy of the text, I appreciate that.

It's always good to re-read docs if you're touching them. Fresh eyes spot problems that were missed in the past, and it's too easy for docs to get outdated. 🙂

**Review by anomiex — APPROVED** [review]

**Comment by anomiex** [comment]
r247584-wpcom

**Pull request merged**